src/selectors/entities/channels.js

```diff
@@ -306,20 +306,33 @@ export const getMembersInCurrentChannel = createSelector(
 export const getUnreads = createSelector(
     getAllChannels,
     getMyChannelMemberships,
-    (channels, myMembers) => {
+    getUsers,
+    getCurrentUserId,
+    (channels, myMembers, users, currentUserId) => {
         let messageCount = 0;
         let mentionCount = 0;
         Object.keys(myMembers).forEach((channelId) => {
             const channel = channels[channelId];
             const m = myMembers[channelId];
+
             if (channel && m) {
+                let otherUserId;
                 if (channel.type === 'D') {
-                    mentionCount += channel.total_msg_count - m.msg_count;
-                } else if (m.mention_count > 0) {
+                    otherUserId = getUserIdFromChannelName(currentUserId, channel.name);
+                    if (users[otherUserId] && users[otherUserId].delete_at === 0) {
```

Comment:
you could instantiate a variable for `users[otherUserId]` instead of accessing the array twice